nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -595,36 +638,178 @@ sub multiLineList {
   # services.openssh.enable = true;
 
   # Open ports in the firewall.
+
   # networking.firewall.allowedTCPPorts = [ ... ];
   # networking.firewall.allowedUDPPorts = [ ... ];
-  # Or disable the firewall altogether.
+
+  # WARNING! WARNING! WARNING! INSECURE!
+  #
+  # To disable the firewall altogether use the following.
+  #
+  # Use only inside trusted networks behind another firewall or else
+  # bad people across the galaxy will be able to access any ports you
+  # leave open.
+  #
   # networking.firewall.enable = false;
 
   # Enable CUPS to print documents.
   # services.printing.enable = true;
 
-  # Enable sound.
+  # Enable ALSA. This option will add ALSA tools to your
+  # systemPackages and make ALSA settings such as volume persistent
+  # across reboots.
+  # Not enabling this doesn't actually disable sound output via ALSA
+  # because ALSA is the default sound API of the Linux kernel.
+
   # sound.enable = true;
+
+  # Configure ALSA sound volume and capture control via common media
+  # keys. This will have no effect unless `sound.enable` is set.
+  # You want to leave this disabled if you run a desktop environment
+  # like KDE, Gnome, Xfce, etc, as those handle such things
+  # themselves.
+  # You might want to enable this if you run a minimalistic desktop
+  # environment or work from bare linux ttys/framebuffers.
+
+  # sound.mediaKeys.enable = true;
+
+  # Enable sound output via PulseAudio daemon.
   # hardware.pulseaudio.enable = true;
 
+  # Note that you don't need to enable PulseAudio daemon to get either
+  # of sound playback or software sound mixing. It will out of the box
+  # over ALSA.
+  #
+  # If you do want to run PulseAudio daemon, however, you may safely
+  # disable `sound.enable`.
+  #
+  # Also note that you can emulate PulseAudio API over ALSA for apps
+  # that refuse to work over ALSA themselves by running them under
+  # apulse. This way you can run one daemon less which benefits your
+  # security (some people claim that PulseAudio codebase lacks in
+  # quality to be connected to untrusted data sources like the browser
+  # by default).
+
+  # environment.systemPackages = [ pkgs.apulse ];
+  # (You might need to merge this line with the above mention of
+  # environment.systemPackages as you can't assign the same attribute
+  # multiple times in a single nix attribute set.)
+
   # Enable the X11 windowing system.
+
   # services.xserver.enable = true;
   # services.xserver.layout = "us";
   # services.xserver.xkbOptions = "eurosign:e";
 
   # Enable touchpad support.
   # services.xserver.libinput.enable = true;
 
```

Comment:
I think all of the changes following here should probably be in the wiki instead.